packages/ui/src/components/va-counter/validateCounter.ts

```diff
@@ -0,0 +1,31 @@
+import { warn } from '../../services/utils'
+
+export const validateCounter = (value: string | number, step: number, min?: number, max?: number) => {
+  const val = Number(value)
+  if (!Number.isNaN(val)) {
+    if (min && max) {
+      if (max < min) {
+        warn(`The maximum value (${max}) can not be less than the minimum value (${min}).`)
+      }
+
+      if (min > max) {
+        warn(`The minimum value (${min}) can not be greater than the maximum value (${max}).`)
+      }
+
+      if ((max - min) % step !== 0) {
+        warn(`Step ${step} is illegal. Counter is non-divisible (Min:Max-${min}:${max}).`)
+      }
```

Comment:
```suggestion
```
we don't need this condition because the counter has some logic to round up to the nearest valid value.
But maybe we should check if the `step` is greater than `max - min`